extras/find_imports.py

```diff
@@ -0,0 +1,136 @@
+                  'gzip', 'hashlib', 'heapq', 'hmac', 'html', 'http', 'imaplib', 'imghdr', 'imp', 
+                  'importlib', 'inspect', 'io', 'ipaddress', 'itertools', 'json', 'keyword', 'lib2to3', 
+                  'linecache', 'locale', 'logging', 'lzma', 'mailbox', 'mailcap', 'marshal', 'math', 
+                  'mimetypes', 'mmap', 'modulefinder', 'msilib', 'msvcrt', 'multiprocessing', 'netrc', 
+                  'nis', 'nntplib', 'numbers', 'operator', 'optparse', 'os', 'ossaudiodev', 'parser', 
+                  'pathlib', 'pdb', 'pickle', 'pickletools', 'pipes', 'pkgutil', 'platform', 'plistlib', 
+                  'poplib', 'posix', 'pprint', 'profile', 'pstats', 'pty', 'pwd', 'py_compile', 'pyclbr', 
+                  'pydoc', 'queue', 'quopri', 'random', 're', 'readline', 'reprlib', 'resource', 
+                  'rlcompleter', 'runpy', 'sched', 'secrets', 'select', 'selectors', 'shelve', 'shlex', 
+                  'shutil', 'signal', 'site', 'smtpd', 'smtplib', 'sndhdr', 'socket', 'socketserver', 
+                  'spwd', 'sqlite3', 'ssl', 'stat', 'statistics', 'string', 'stringprep', 'struct', 
+                  'subprocess', 'sunau', 'symbol', 'symtable', 'sys', 'sysconfig', 'syslog', 'tabnanny', 
+                  'tarfile', 'telnetlib', 'tempfile', 'termios', 'test', 'textwrap', 'threading', 'time', 
+                  'timeit', 'tkinter', 'token', 'tokenize', 'trace', 'traceback', 'tracemalloc', 'tty', 
+                  'turtle', 'turtledemo', 'types', 'typing', 'unicodedata', 'unittest', 'urllib', 'uu', 
+                  'uuid', 'venv', 'warnings', 'wave', 'weakref', 'webbrowser', 'winreg', 'winsound', 
+                  'wsgiref', 'xdrlib', 'xml', 'xmlrpc', 'zipapp', 'zipfile', 'zipimport', 'zlib', 
+                  'zoneinfo'])
+
+# find modules that are defined inside the NEST code base
+pn_modules = set(os.path.splitext(os.path.split(fn)[-1])[0] 
+                 for fn in glob.glob(f'{source_dir}/**/*.py', recursive=True))
+
+# exclude modules from the Python Standard Library and modules defined within NEST
+exclude_modules = py_modules.union(pn_modules)
+
+# packages forming the base part of NEST
+base_dirs = {f'{source_dir}/pynest/nest', f'{source_dir}/pynest/examples',
+             f'{source_dir}/pynest/tests', f'{source_dir}/testsuite'}
+
+# server package
+server_dir = f'{source_dir}/pynest/nest/server'
+
+# Match for import lines, picks up only top-level package and ignores relative imports
+# We need the \s* at the beginning because some imports are nested in try-blocks. This
+# can lead to false positives if a comment line or multiline string line begins with
+# "import" or "from".
+import_re = re.compile('\s*(import|from)\s+(\w+)')
+
+imports = defaultdict(set)
+for dirpath, _, fnames in os.walk(source_dir):
+
+    if any([exclude_dir in dirpath for exclude_dir in exclude_dirs]):
+        continue
+
+    for fname in fnames:
+        if not fname.endswith('.py'):
```

Comment:
Do we also have to scan [`pynestkernel.pxd`](https://github.com/nest/nest-simulator/blob/master/pynest/pynestkernel.pxd) and [`pynestkernel.pyx`](https://github.com/nest/nest-simulator/blob/master/pynest/pynestkernel.pyx)?

Reply:
They do not include anything not included in Cython or "the base" and I assume they will stay very stable, so I left them out to avoid complicating the test here.

Reply:
Alright.